src/nft/hooks/usePermit2Approval.ts

```diff
@@ -0,0 +1,44 @@
+import { sendAnalyticsEvent } from '@uniswap/analytics'
+import { InterfaceEventName } from '@uniswap/analytics-events'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { UNIVERSAL_ROUTER_ADDRESS } from '@uniswap/universal-router-sdk'
+import { useWeb3React } from '@web3-react/core'
+import usePermit2Allowance, { AllowanceState } from 'hooks/usePermit2Allowance'
+import { useCallback, useMemo, useState } from 'react'
+import invariant from 'tiny-invariant'
+
+export default function usePermit2Approval(amount?: CurrencyAmount<Token>, maximumAmount?: CurrencyAmount<Token>) {
```

Comment:
Does web not already have a hook for approving the use of Permit2?

Reply:
they don't this is the code from web reduced to a hook